packages/neuron-ui/src/components/History/TablePagination.tsx

```diff
@@ -2,29 +2,67 @@ import React from 'react'
 import { Pagination } from 'react-bootstrap'
 
 interface TablePaginationProps {
-  page: number
+  currentPage: number
   pageSize: number
   total: number
+  displayCount?: number
   onChange: (page: number) => void
 }
 
-const TablePagination = ({ total, page, pageSize, onChange }: TablePaginationProps) => (
+const PageNos = ({
+  currentPage,
+  pageNos,
+  pageCount,
+  onChange,
+}: {
+  currentPage: number
+  pageNos: number[]
+  pageCount: number
+  onChange: (pageNo: number) => void
+}) => (
   <Pagination>
-    <Pagination.First />
-    <Pagination.Prev />
-    {Array.from({
-      length: Math.round(total / pageSize),
-    }).map((_value: any, pageNo: number) => {
+    <Pagination.First onClick={() => onChange(0)} />
+    <Pagination.Prev disabled={currentPage === 0} onClick={() => onChange(currentPage - 1)} />
+    {pageNos.map(pageNo => {
+      if (pageNo < 0) {
+        return <Pagination.Ellipsis disabled />
+      }
       return (
-        <Pagination.Item key={Math.random()} active={pageNo === page} onClick={() => onChange(pageNo)}>
+        <Pagination.Item key={Math.random()} active={pageNo === currentPage} onClick={() => onChange(pageNo)}>
           {pageNo + 1}
         </Pagination.Item>
       )
     })}
-
-    <Pagination.Next />
-    <Pagination.Last />
+    <Pagination.Next disabled={currentPage === pageCount - 1} onClick={() => onChange(currentPage + 1)} />
+    <Pagination.Last onClick={() => onChange(pageCount - 1)} />
   </Pagination>
 )
 
+const TablePagination = ({ total, currentPage, pageSize, displayCount = 5, onChange }: TablePaginationProps) => {
```

Comment:
I feel this component deserves its own file under `widgets`.

Reply:
Yes, I'll do it in another PR